ideascube/monitoring/forms.py

```diff
@@ -149,3 +151,37 @@ def clean_loan(self):
             # Should araise only after migration 0.3.2 to 0.3.0.
             loan = Loan.objects.due().filter(specimen__barcode=barcode).first()
         return loan
+
+
+class StockImportForm(forms.Form):
+    source = forms.FileField(label=_('CSV File'), required=True)
+
+    def save(self):
+        source = self.cleaned_data['source']
+        source = TextIOWrapper(source.file, encoding='utf-8')
+        items = []
+        errors = []
+
+        for index, row in enumerate(csv.DictReader(source)):
+            try:
+                data = {
+                    'module': row['module'], 'name': row['name'],
+                    'description': row['description'],
+                }
+
+            except KeyError as e:
+                errors.append(_('Missing column "{}" on line {}').format(
+                    e.args[0], index + 1))
+                continue
+
+            try:
+                item = StockItem(**data)
+                item.save()
+                items.append(item)
```

Comment:
I'd move this line to an `else` block, as it's supposed to raise any exception we want to catch here.